pkg/kubelet/kubelet.go

```diff
@@ -2922,6 +2921,14 @@ func (kl *Kubelet) ListenAndServeReadOnly(address net.IP, port uint) {
 	server.ListenAndServeKubeletReadOnlyServer(kl, kl.resourceAnalyzer, address, port, kl.containerRuntime)
 }
 
+func (kl *Kubelet) cleanUpContainersInPod(podId types.UID, exitedContainerID string) {
+	if podStatus, err := kl.podCache.Get(podId); err == nil {
+		if status, ok := kl.statusManager.GetPodStatus(podId); ok {
+			kl.containerDeletor.deleteContainersInPod(exitedContainerID, podStatus, status.Phase == api.PodFailed)
```

Comment:
This just happened to be right, because our test doesn't test getting container log after pod failed.

If we start a pod:
```yaml
apiVersion: v1
kind: Pod
metadata:
 name: busyloop
spec:
 restartPolicy: Never
 containers:
 - name: busyloop
   image: busybox:1.24
   resources:
    limits:
     cpu: "1000m"
    requests:
     cpu: "1000m"
   command:
   - "/bin/sh"
   - "-c"
   - "echo hello; exit 1"
```

With this change, we'll get:
```console
$ k logs busyloop
Error from server: Error response from daemon: no such id: 1e37f1c8b8b86620170cf4bad76d877659473cdc4c04fd78f1b3ad407eb95f55
```

Without this change, we'll get:
```
$ k logs busyloop
hello
```